### PR TITLE
platform(general): remove igraph

### PR DESCRIPTION
**Pull request by lirshindalman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Removed `igraph` dependency and every usage from checkov

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by omryMen — APPROVED** [review]

File: checkov/common/images/graph/image_referencer_provider.py [hunk 1]
```diff
@@ -26,16 +24,14 @@ def __init__(self, graph_connector: LibraryGraph,
                      str, _ExtractImagesCallableAlias]):
         self.graph_connector = graph_connector
         self.supported_resource_types = supported_resource_types
-        self.graph_framework = os.environ.get('CHECKOV_GRAPH_FRAMEWORK', 'IGRAPH')
+        self.graph_framework = os.environ.get('CHECKOV_GRAPH_FRAMEWORK', 'NETWORKX')
```
Comment (lirshindalman) [inline checkov/common/images/graph/image_referencer_provider.py:27]: maybe change the default to 'RUSTWORKX'?

File: checkov/common/util/env_vars_config.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def __init__(self) -> None:
         self.GITHUB_CONFIG_FETCH_DATA = convert_str_to_bool(os.getenv("CKV_GITHUB_CONFIG_FETCH_DATA", True))
         self.GITLAB_CONF_DIR_NAME = os.getenv("CKV_GITLAB_CONF_DIR_NAME", "gitlab_conf")
         self.GITLAB_CONFIG_FETCH_DATA = convert_str_to_bool(os.getenv("CKV_GITLAB_CONFIG_FETCH_DATA", True))
-        self.GRAPH_FRAMEWORK = os.getenv("CHECKOV_GRAPH_FRAMEWORK", "IGRAPH")
+        self.GRAPH_FRAMEWORK = os.getenv("CHECKOV_GRAPH_FRAMEWORK", "NETWORKX")
```
Comment (lirshindalman) [inline checkov/common/util/env_vars_config.py:42]: the same here

**Review by SteveVaknin — APPROVED** [review]
Looks good, Agree with @lirshindalman about changing the defaults 👍

File: checkov/common/images/graph/image_referencer_provider.py (hunk 1 above)
Comment (bo156) [reply checkov/common/images/graph/image_referencer_provider.py:27]: Done

File: checkov/common/util/env_vars_config.py (hunk 2 above)
Comment (bo156) [reply checkov/common/util/env_vars_config.py:42]: done

**Pull request merged**